components/bl-mapbox-component/src/index.js

```diff
@@ -0,0 +1,230 @@
+import { useEffect, useState, useRef } from 'react';
+import mapboxgl from './lib/mapbox';
+import MapboxGeocoder from './lib/mapbox-geocoder';
+import MapboxDirections from './lib/mapbox-directions';
+import { createActions } from './actions';
+
+const defaultValues = {
+  START_POS           : {
+    lat: 0,
+    lng: 0
+  },
+  MAP_STYLE           : 'mapbox://styles/mapbox/streets-v11',
+  ZOOM                : 10,
+  PROJECTION          : 'mercator',
+  LOWER_ATMOSPHERE    : '#BAD2EB',
+  UPPER_ATMOSPHERE    : '#245CDF',
+  ATMOSPHERE_THICKNESS: 0.2,
+  SPACE_COLOR         : '#0B0B19',
+  STAR_INTENSITY      : 0.2
+};
+
+export default function Mapbox({ component, eventHandlers }) {
+  const mapContainer = useRef(null);
+  const mapRef = useRef(null);
+
+  const [polygonsObject, setPolygonsObject] = useState([]);
+
+  const { onClick, onMarkerClick, onPolygonClick, onDeterminingGeoposition } = eventHandlers;
+
+  const {
+    markers,
+    polygons,
+    startPos,
+    zoom,
+    fullScreen,
+    accessToken,
+    projection,
+    lowerAtmosphere,
+    upperAtmosphere,
+    atmosphereThickness,
+    spaceColor,
+    starIntensity,
+    mapStyle,
+    navigation,
+    searchBar,
+    geolocation,
+    directions
+  } = component;
+
+  if (!display) {
+    return null;
+  }
+
+  useEffect(() => {
+    if (polygons) {
+      if (polygons.length) {
+        mapRef.current.on('load', () => {
+          polygonsObject.map(el => {
+            if (mapRef.current.getSource(el.id)) {
```

Comment:
In my opinion, it is better to move `mapRef.current.getSource(el.id) ` to separate variable.